### PR TITLE
(PA-312) Add sles-12-s390x as a platform

### PR DESCRIPTION
**Pull request by ScottGarman**
This is a zLinux platform for SLES 12 running on IBM's s390x arch.

### CODE REVIEW

**Comment by ScottGarman** [comment]
Running an ad-hoc jenkins build #125. https://jenkins.puppetlabs.com/view/puppet-agent/view/ad%20hoc/job/platform_puppet-agent_init-van-int_suite-manual-ad-hoc/125/

**Comment by ScottGarman** [comment]
Found some issues thanks to that jenkins build, so here's one that completed successfully: https://jenkins.puppetlabs.com/view/puppet-agent/view/ad%20hoc/job/platform_puppet-agent_init-van-int_suite-manual-ad-hoc/128/

**Comment by McdonaldSeanp** [comment]
All PRs should wait until we get a passing PA master again

**Comment by shrug** [comment]
Adding do not merge labels until we get a clean ci run

**Comment by ScottGarman** [comment]
ad-hoc jenkins job on solaris targets that passed: https://jenkins.puppetlabs.com/view/puppet-agent/view/ad%20hoc/job/platform_puppet-agent_pkg-van-ship_manual-ad-hoc/129/

**Comment by ScottGarman** [comment]
and a bunch of linux-based targets, cross-compiled and otherwise: https://jenkins.puppetlabs.com/view/puppet-agent/view/ad%20hoc/job/platform_puppet-agent_pkg-van-ship_manual-ad-hoc/130/

File: configs/components/libxslt.rb [hunk 1]
```diff
@@ -31,7 +31,7 @@
     pkg.environment "CFLAGS" => settings[:cflags]
   end
 
-  if platform.name =~ /huaweios|solaris-11/ || platform.architecture == "s390x"
+  if platform.is_cross_compiled_linux? || platform.name =~ /solaris-11/
```
Comment (stahnma) [inline configs/components/libxslt.rb]: Can we comment on why we can't use is_cross_compiled? here?

**Pull request merged**